### PR TITLE
PhysicsNeMo-Mesh: fix silent data drop in `Mesh(cell_data=pv_mesh.cell_data, ...)`

### PR DESCRIPTION
**Pull request by peterdsharpe**
## Description

Under `tensordict >= 0.12`, constructing a `Mesh` directly from a PyVista mesh's data containers silently dropped all keys:

```python
from physicsnemo.mesh.mesh import Mesh
import pyvista as pv

pv_mesh = pv.read("freestream.vtp")  # has cell_data["U"]
mesh = Mesh(points=..., cells=..., cell_data=pv_mesh.cell_data)
assert "U" in mesh.cell_data           # FAILS - mesh.cell_data is empty
```

Instantiating directly from PyVista mesh data containers was never *officially* supported by the `Mesh` constructor API (i.e., a good type-checker would yell at you to `dict()` wrap it if you tried it) - but it's the kind of thing that an end-user might do cavalierly, and expect it to work.

This PR fixes that. After the fix, all entries from `pv_mesh.cell_data` (and `point_data`, `field_data`) survive direct `Mesh(...)` construction.

The matching `from_pyvista` flow was already fixed in PR #1591 by pre-converting to a plain `dict[str, torch.Tensor]` before calling `Mesh(...)`. This PR closes the remaining hole for users that build a `Mesh` directly from a `pyvista` mesh's data containers.

## Root cause

`Mesh` is declared `@tensorclass(tensor_only=True, shadow=True)`. With `tensor_only=True` and no custom `__setattr__` on `Mesh`, the `tensorclass` decorator in `tensordict >= 0.12` replaces the user-defined `Mesh.__init__` with an auto-generated fast-path that writes each kwarg directly to the underlying `_tensordict` via its internal `_validate_value`. For values that are non-dict `Mapping`s (PyVista's `DataSetAttributes` is one), `_validate_value` silently wraps the value as `NonTensorData(data=<original Mapping>)` instead of converting it to a `TensorDict`. From the outside, the field looks like an empty `NonTensorData` with zero keys.

The dict-coercion logic that already lives in `Mesh.__init__` is, in this regime, **dead code**: the auto-init never calls user `__init__`. By the time `__post_init__` runs, the original `Mapping` has already been buried inside a `NonTensorData` wrapper.

For background on when the auto-init bypasses user `__init__`, see the `_init_wrapper` logic in `tensordict/tensorclass.py` (the wrapper takes the slow `__init__`-call path only when the class also defines `__setattr__`).

## Fix

`Mesh.__post_init__` now unwraps any `NonTensorData` it finds on `point_data` / `cell_data` / `global_data` before coercing the contents to `TensorDict`:

```python
for field_name, batch_size in (
    ("point_data", torch.Size([self.n_points])),
    ("cell_data",  torch.Size([self.n_cells])),
    ("global_data", torch.Size([])),
):
    value = getattr(self, field_name)
    if isinstance(value, TensorDict):
        value.batch_size = batch_size
        continue
    if isinstance(value, NonTensorData):
        value = value.data  # extract original Mapping from fast-path wrapper
    setattr(
        self,
        field_name,
        TensorDict(
            {} if value is None else dict(value),
            batch_size=batch_size,
            device=self.points.device,
        ),
    )
```

This is the only change to `physicsnemo/mesh/mesh.py` besides adding `NonTensorData` to the existing `tensordict` import. The three previously- duplicated normalization blocks for `point_data`, `cell_data`, and `global_data` collapse into the loop above.

`NonTensorData.data` is a documented public field of `tensordict.NonTensorDataBase` ("A base class to carry non-tensor data."), so this does not depend on tensordict private API.

## Why not just drop `tensor_only=True`?

I considered that, and it does fix the symptom (with `tensor_only=False` the auto-init's slow path stores the `Mapping` as-is, so `__post_init__` can convert it). But it costs ~10 us per `Mesh` construction (microbenchmark on this hardware: 9.8 us with the chosen fix, 22.6 us with `tensor_only=False`), a 2.3x slowdown on the construction path. `Mesh` is constructed in a number of inner operations (slicing, transforms, batch concatenation, `from_pyvista`), so it felt worth keeping the fast path. A 4-line `isinstance(value, NonTensorData)` branch in `__post_init__` recovers the dropped `Mapping` without giving up the fast-path speed.

## Tests

Three new test groups, all passing with the fix:

1. **`TestMeshConstructorAcceptsPyVistaDataSetAttributes`** in `test/mesh/io/io_pyvista/test_data_preservation.py` - 4 cases for direct `Mesh(...)` construction with `pv_mesh.cell_data`, `point_data`, `field_data`, and all three at once. All four fail on `main` before this PR and pass after.

2. **`test_with_cell_and_global_data`** in `test/mesh/mesh/test_serialization.py` - closes a pre-existing coverage gap by adding `cell_data` and `global_data` to the existing memmap round-trip suite, which previously only exercised `point_data`.

3. **`TestPmshGoldenFixture`** in `test/mesh/mesh/test_pmsh_golden.py` - cross-version compatibility check that loads a committed `.pmsh` fixture (`test/mesh/mesh/golden_pmsh/v2.0_two_triangles.pmsh/`, 16 files / ~1.7 KB) and asserts every field round-trips exactly. The fixture was generated by the *pre-fix* `Mesh` code via the companion `_regenerate.py` script, so this test empirically confirms the fix does not change the on-disk `.pmsh` format. Future PRs that quietly alter the layout (renaming a tensorclass field, dropping `shadow=True`, swapping the decorator for `class Mesh(TensorClass)`, etc.) will fail this test loudly. Re-run `_regenerate.py` and commit the resulting tree only when the format intentionally changes.

The full mesh suite (now 1854 tests, +4 from this PR) continues to pass.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: test/mesh/mesh/test_pmsh_golden.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import torch
+
+from physicsnemo.mesh.mesh import Mesh
+
+### Locate the regeneration helper without requiring `golden_pmsh` to be a package.
+_REGEN_PATH = Path(__file__).parent / "golden_pmsh" / "_regenerate.py"
+_spec = importlib.util.spec_from_file_location("_pmsh_golden_regen", _REGEN_PATH)
+assert _spec is not None and _spec.loader is not None
+_regen = importlib.util.module_from_spec(_spec)
+_spec.loader.exec_module(_regen)
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply test/mesh/mesh/test_pmsh_golden.py:57]: Fixed

**Review by ktangsali — APPROVED** [review]
LGTM!
